web-src/src/components/GenerateButton.js

```diff
@@ -74,12 +74,22 @@ export function GenerateButton() {
     sampleRUM('genai:prompt:generate', { source: 'GenerateButton#handleGenerate' });
     setGenerationInProgress(true);
     setIsOpenPromptEditor(false);
+    const startTime = Date.now();
     generateResults()
       .catch((error) => {
         ToastQueue.negative(error.message, { timeout: 2000 });
+
+        // capture network errors
+        if (error.status > 300) {
+          log('prompt:generate:networkerror', { status: error.status, message: error.message });
+        }
       })
       .finally(() => {
         setGenerationInProgress(false);
+
+        const endTime = Date.now();
+        const responseTime = ((endTime - startTime) / 1000).toFixed(2);
```

Comment:
Please add comments like 
 const responseTime = ((endTime - startTime) / 1000).toFixed(2);// response time in seconds or use it in variable name like responseTimeSeconds